src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+
+  getArea(): Number;
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number = 0,
+    public b: number = 0,
+    public c: number = 0,
+    public shape: Shape = 'triangle',
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Error');
+    }
 
-}
+    const longest = Math.max(+this.a, +this.b, +this.c);
```

Comment:
Sides are already numbers.

```suggestion
    const longest = Math.max(this.a, this.b, this.c);
```